### PR TITLE
API: deprecate warning for show_in_browser

### PR DESCRIPTION
**Pull request by MridulS**
Follow up from https://github.com/astropy/astropy/pull/15905

Being a bit bold here, let's add a pending deprecation warning to `show_in_browser`. By removing `show_in_browser` we can get rid of the JS code in astropy.

We can use https://github.com/astropy/astropy/issues/16067 to solicit feedback.

### CODE REVIEW

**Comment by taldcroft** [comment]
@MridulS - there is a merge conflict that needs to be resolved. Otherwise this looks good to me.

**Review by taldcroft — APPROVED** [review]
Looks good!

**Comment by taldcroft** [comment]
I set this for auto-merge, but it looks like the milestone_checker is stuck. hmm.

**Comment by neutrinoceros** [comment]
@taldcroft setting a milestone should retrigger and fix it (🤞🏻)

**Comment by taldcroft** [comment]
@neutrinoceros - do'oh!! thanx.

File: astropy/table/table.py [hunk 1]
```diff
@@ -1841,6 +1841,13 @@ def show_in_notebook(
         html += jsv.ipynb(tableid, css=css, sort_columns=sortable_columns)
         return HTML(html)
 
+    @deprecated(
+        "6.1",
+        pending=True,
```
Comment (eerovaher) [inline astropy/table/table.py:1846]: Why is this not a proper deprecation?

Comment (taldcroft) [reply astropy/table/table.py:1846]: I think the idea was to take this one a bit slower to gather user inputs.

Comment (pllim) [reply astropy/table/table.py:1846]: I don't think most user would see `pending=True` warning.  Could you please open a follow-up issue on when you think we can remove `pending=True` here? Thanks!

Comment (pllim) [reply astropy/table/table.py:1846]: Might also want to post to astropy-dev once you have this follow-up issue. Any further discussions on a merged PR would be easily overlooked.

**Pull request merged**